src/test.ts

```diff
@@ -0,0 +1,176 @@
+import os from 'os';
+import S3 from './S3';
+import axios from 'axios';
+import Redis from './Redis';
+import urlParser from 'url';
+import logger from './Logger';
+import { config } from './config';
+import MediaWiki from './MediaWiki';
+import Downloader from './Downloader';
+import fs, { readFileSync } from 'fs';
+import * as QueryStringParser from 'querystring';
+import { isValidEmail, ensureTrailingChar } from './util';
+
+function IsURL(url: string) {
+  const strRegex =
+    '^((https|http|ftp|rtsp|mms)?://)' +
+    '?(([0-9a-z_!~*\'().&=+$%-]+: )?[0-9a-z_!~*\'().&=+$%-]+@)?' +
+    '(([0-9]{1,3}.){3}[0-9]{1,3}' +
+    '|' +
+    '([0-9a-z_!~*\'()-]+.)*' +
+    '([0-9a-z][0-9a-z-]{0,61})?[0-9a-z].' +
+    '[a-z]{2,6})' +
+    '(:[0-9]{1,4})?' +
+    '((/?)|' +
+    '(/[0-9a-z_!~*\'().;?:@&=+$,%#-]+)+/?)$';
+  const re = new RegExp(strRegex);
+  return re.test(url);
+}
+
+export async function test_all(argv: any) {
+  // extracting all arguments
+  const {
+    speed: _speed,
+    adminEmail,
+    mwUrl,
+    mwWikiPath,
+    mwApiPath,
+    mwRestApiPath,
+    mwModulePath,
+    mwDomain,
+    mwUsername,
+    mwPassword,
+    requestTimeout,
+    format,
+    filenamePrefix,
+    publisher: _publisher,
+    outputDirectory: _outputDirectory,
+    addNamespaces: _addNamespaces,
+    articleList: _articleList,
+    customZimFavicon,
+    optimisationCacheUrl,
+    noLocalParserFallback,
+    forceLocalParser,
```

Comment:
This is a duplicate of code which should avoid to define the argument line parsing twice. We should have shared code.

Reply:
@kelson42 I don't understand how to do this part....can you help?

Reply:
OK, maybe we should leave that for the moment.